### ISSUE TITLE
Integrate IBM Granite Tiny Time Mixer (TTM-R1 and TTM-R2) Model for Time Series Forecasting

### ISSUE THREAD

**Issue by Killer3048** [autogluon/autogluon#4740]
### Description

I propose adding support for the **IBM Granite Tiny Time Mixer (TTM)** model to the `timeseries` module of **AutoGluon**. TTM is a lightweight, pre-trained model specifically designed for multivariate time series forecasting, featuring fewer than 1 million parameters. Despite its compact size, TTM delivers superior performance in zero-shot and few-shot forecasting tasks compared to larger models with billions of parameters.
Integrating TTM into AutoGluon would provide users with a state-of-the-art yet computationally efficient option for forecasting, enhancing the library's capabilities and appeal to researchers and practitioners. This integration would allow users to leverage TTM's strengths directly within the AutoGluon framework, facilitating seamless model training, tuning, and deployment.

### Benefits of Integration

- **Efficiency**: TTM's small size ensures low computational overhead, enabling deployment on resource-constrained environments such as laptops or single GPU machines.
- **Performance**: Outperforms several benchmarks in zero-shot and few-shot scenarios, offering high accuracy with minimal training data.
- **Flexibility**: Supports various forecasting horizons and context lengths, making it adaptable to diverse time series forecasting tasks.
- **Ease of Use**: Seamless integration with AutoGluon's existing `TimeSeriesPredictor` API, allowing users to easily incorporate TTM into their workflows.

### Mock Example of Integration

Below is a mock example demonstrating how TTM could be integrated into AutoGluon's existing workflow using the `TimeSeriesPredictor` class.

```python
import pandas as pd
from autogluon.timeseries import TimeSeriesDataFrame, TimeSeriesPredictor

# Load training data
df = pd.read_csv("https://autogluon.s3.amazonaws.com/datasets/timeseries/m4_hourly_subset/train.csv")
print("First 5 rows of the training data:")
print(df.head())

# Prepare data in TimeSeriesDataFrame format
train_data = TimeSeriesDataFrame.from_data_frame(
    df,
    id_column="item_id",
    timestamp_column="timestamp"
)
print("\nFirst 5 rows of the TimeSeriesDataFrame:")
print(train_data.head())

# Create a TimeSeriesPredictor object with TTM integrated
predictor = TimeSeriesPredictor(
    prediction_length=48,             # Number of steps to forecast into the future
    path="autogluon-m4-hourly",        # Directory path to save trained models
    target="target",                   # Name of the target column
    eval_metric="MASE",                # Evaluation metric (Mean Absolute Scaled Error)
    hyperparameters={
        'tiny_time_mixer': {           # Name of the model as integrated in AutoGluon
            'context_length': 512,     # Length of historical data used for forecasting
            'forecast_length': 48,     # Length of the forecast horizon
            # Additional hyperparameters specific to TTM can be specified here
            'learning_rate': 0.001,
            'num_epochs': 20,
        }
    }
)

# Train the predictor, including Tiny Time Mixer in hyperparameters
predictor.fit(
    train_data,
    presets="medium_quality",          # Preset configurations for training quality
    time_limit=600,                     # Time limit for training in seconds (10 minutes)
)

# Load test data
test_df = pd.read_csv("https://autogluon.s3.amazonaws.com/datasets/timeseries/m4_hourly_subset/test.csv")
test_data = TimeSeriesDataFrame.from_data_frame(
    test_df,
    id_column="item_id",
    timestamp_column="timestamp"
)

# Generate forecasts
predictions = predictor.predict(test_data)
print("\nFirst 5 rows of the predictions:")
print(predictions.head())

# Display the leaderboard to evaluate model performance on test data
leaderboard = predictor.leaderboard(test_data)
print("\nLeaderboard of models:")
print(leaderboard)
```

### Implementation Steps

1. **Model Wrapper Creation**:
   - Develop a wrapper for TTM that adheres to AutoGluon's `AbstractTimeSeriesModel` interface.
   - Ensure compatibility with AutoGluon's training, prediction, and evaluation workflows.

2. **Hyperparameter Integration**:
   - Define TTM-specific hyperparameters within AutoGluon's hyperparameter tuning framework.
   - Allow users to customize TTM's `context_length`, `forecast_length`, and other relevant parameters.

3. **Model Registration**:
   - Register TTM as a viable model option within AutoGluon's `TimeSeriesPredictor`.
   - Ensure it appears alongside existing models like ARIMA, ETS, DeepAR, etc.

4. **Documentation and Examples**:
   - Provide comprehensive documentation on using TTM within AutoGluon.
   - Include example notebooks demonstrating training, tuning, and forecasting with TTM.

5. **Testing and Validation**:
   - Conduct thorough testing to ensure TTM's integration does not disrupt existing functionalities.
   - Validate performance gains and efficiency improvements through benchmarking against existing models.

### References

- [IBM Granite Time Series Documentation](https://www.ibm.com/granite/docs/models/time-series/)
- [IBM Granite Time Series GitHub Repository](https://github.com/ibm-granite/granite-tsfm)
- [IBM Granite TTM-R1 Model on Hugging Face](https://huggingface.co/ibm-granite/granite-timeseries-ttm-r1)
- [IBM Granite TTM-R2 Model on Hugging Face](https://huggingface.co/ibm-granite/granite-timeseries-ttm-r2)
- [Research Paper: Tiny Time Mixers (TTMs)](https://arxiv.org/abs/2401.03955)
### Conclusion

Integrating IBM Granite's Tiny Time Mixer (TTM) into AutoGluon's `timeseries` module would significantly enhance the library's forecasting capabilities. TTM's efficiency and superior performance in zero-shot and few-shot scenarios make it an invaluable addition, providing users with more options to achieve accurate and reliable time series forecasts with minimal computational resources.
